pkg/kubelet/util/pluginwatcher/plugin_watcher.go

```diff
@@ -158,67 +176,98 @@ func (w *Watcher) invokeRegistrationCallbackAtHandler(ctx context.Context, clien
 	}); err != nil {
 		return fmt.Errorf("failed to send registration status at socket %s, err: %v", socketPath, err)
 	}
+
 	chanForAckOfNotification <- true
 	return nil
 }
 
+// Handle filesystem notify event, it sends errors to error channel and return the errors.
+func (w *Watcher) handlefsnotifyEvent(event fsnotify.Event) error {
+	if event.Op&fsnotify.Create == fsnotify.Create {
+		fi, err := os.Stat(event.Name)
+		if err != nil {
+			w.errorCh <- fmt.Errorf("stat file %s failed: %v", event.Name, err)
```

Comment:
You should just return the error and log it in the caller function.
It would remove this `errorChan`

Reply:
addressed in the update, thanks